joblib/test/test_memory.py

```diff
@@ -126,10 +126,16 @@ def f(l):
             # happens if the directory disappears
             shutil.rmtree(env['dir'], ignore_errors=True)
             g = memory.cache(f)
-            g(1)
-            g.clear(warn=False)
-            current_accumulator = len(accumulator)
-            out = g(1)
+            if compress is True and mmap_mode == 'r':
+                # Should raise an exception
+                nose.tools.assert_raises(ValueError, g, 1)
```

Comment:
It seems like a controversial change in the Memory object. Maybe we want to stick with the warnings as previously and possibly tackle this in a separate PR.